### PR TITLE
Abstract ControlScheme and Javadoc

### PR DESCRIPTION
**Pull request by isaaclloyd**
I wrote an abstract class ControlScheme for control scheme classes to inherit from.  I also wrote Javadoc for everything.

### CODE REVIEW

**Review by alex-born — CHANGES_REQUESTED** [review]
My comments here were pretty light. I would like @MaxThePrisberry to look all these comments and make sure they accurately describe what is happening. 

A few commenting notes, you don't need to comment this when they are obvious. This will save you some time.

An example is this line of code:

`private WPI_TalonSRX leftMasterTalon;`

or 

`private double speed;`

This is pretty self explanatory. Make sure all comments add information that the code does not convey or explain pieces of code that are difficult to understand. An example that might need to be enumerated more is the drive signal. A new person might ask the following questions:
- What is a drive signal
- Why do we need this class?

Additionally if you are using a common design pattern, just note that once at the top of your doc.

When commenting code, explain the content in plain english. You shouldn't need to explain it with code unless you are using an example. You are illuminating function with javadoc, not how the class works. If you want to explain how a complicated method works, use inline documentation.

When using javadoc, there are special annotations. [See tutorialspoint.](https://www.tutorialspoint.com/java/java_documentation.htm) The `@param`, `@return` and `@throws` annotations are the most important. 

I should have talked to you about this ahead of time. Sorry. Keep up the good work! Can't wait to merge this!

**Review by MaxThePrisberry — CHANGES_REQUESTED** [review]
We need to restore these imports so the robot can run. Otherwise, it's looking good.

**Comment by nprisbrey** [comment]
For outdated code.
